types/mongoose-aggregate-paginate-v2/index.d.ts

```diff
@@ -0,0 +1,90 @@
+//                 simonxca <https://github.com/simonxca>
+//                 woutgg <https://github.com/woutgg>
+//                 oktapodia <https://github.com/oktapodia>
+//                 Dongjun Lee <https://github.com/ChazEpps>
+//                 gamsterX <https://github.com/gamsterx>
+//                 knyuwork <https://github.com/knyuwork>
+//                 LiRen Tu <https://github.com/tuliren>
+//                 Alexandre Croteau <https://github.com/acrilex1>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// Minimum TypeScript Version: 3.2
+//
+// Based on type declarations for mongoose-paginate-v2 1.3.
+
+declare module 'mongoose' {
+    interface CustomLabels {
+        totalDocs?: string;
+        limit?: string;
+        page?: string;
+        totalPages?: string;
+        docs?: string;
+        nextPage?: string;
+        prevPage?: string;
+        pagingCounter?: string;
+        hasPrevPage?: string;
+        hasNextPage?: string;
+    }
+
+    interface PaginateOptions {
+        /* tslint:disable-next-line: ban-types */
+        sort?: Object | string;
+        offset?: number;
+        page?: number;
+        limit?: number;
+        customLabels?: CustomLabels;
+        /* If pagination is set to `false`, it will return all docs without adding limit condition. (Default: `true`) */
+        pagination?: boolean;
+        allowDiskUse?: boolean;
+        countQuery?: object;
+    }
+
+    interface QueryPopulateOptions {
+        /** space delimited path(s) to populate */
+        path: string;
+        /** optional fields to select */
+        select?: any;
+        /** optional query conditions to match */
+        match?: any;
+        /** optional model to use for population */
+        model?: string | Model<any>;
+        /** optional query options like sort, limit, etc */
+        options?: any;
+        /** deep populate */
+        populate?: QueryPopulateOptions | QueryPopulateOptions[];
+    }
+
+    interface AggregatePaginateResult<T> {
+        docs: T[];
+        totalDocs: number;
+        limit: number;
+        page?: number;
+        totalPages: number;
+        nextPage?: number | null;
+        prevPage?: number | null;
+        pagingCounter: number;
+        hasPrevPage: boolean;
+        hasNextPage: boolean;
+        meta?: any;
+        [customLabel: string]: T[] | number | boolean | null | undefined;
+    }
+
+    interface AggregatePaginateModel<T extends Document> extends Model<T> {
+        aggregatePaginate(
+            query?: Aggregate<T[]>,
+            options?: PaginateOptions,
+            callback?: (err: any, result: AggregatePaginateResult<T>) => void,
+        ): Promise<AggregatePaginateResult<T>>;
+    }
+
+    function model(name: string, schema?: Schema, collection?: string, skipInit?: boolean): AggregatePaginateModel<any>;
+}
+
+import mongoose = require('mongoose');
+declare function _(schema: mongoose.Schema): void;
+export = _;
```

Comment:
I have to admit that "_" is not my favorite identifier.

Reply:
No problem, what identifier do you prefer? Once again, this was copied from mongoose-paginate-v2 and I don't see anything in the Hanbook of the README discouraging this practice. In fact, it is event used by multiple other packages when a function has no name in the original package (default export). Personally, I would find it weird to add a name for a function that is the default export of a package, because the name as no meaning upstream and could be missleading if used elsewhere.

Reply:
A lot of people use the name under which it's going to be imported (e.g. "mongooseAggregatePaginate" in the tests below).  But even "ap" would be easier to read.  I certainly wouldn't block the PR for this.

Reply:
Thanks for your feedback, fixed!